src/app/main.rs

```diff
@@ -65,7 +65,7 @@ struct ProxyParts<G> {
     trace_level: trace::LevelHandle,
 
     admin_listener: Listen<identity::Local, ()>,
-    control_listener: Option<Listen<identity::Local, ()>>,
+    control_listener: Option<(Listen<identity::Local, ()>, identity::Name)>,
```

Comment:
I kept this as a tuple because we destructure it right away [here](https://github.com/linkerd/linkerd2-proxy/pull/295/files#diff-28081f5c7594c599dfa097ac3ddd0193R395). I could be convinced to add a similar `TapSettings` type in `tap.rs`, but after trying that out it seemed like a lot of boilerplate for not much of an abstraction.

Reply:
Yep I should have pointed that out too. For the clarity we gain in that access though, it still seemed a little unnecessary to add a new struct for it.